### DISCUSSION TITLE
reducing duplicates x-transition

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#2418] [1. Help] [answered]
Hi All, how can i reduce duplicate x-transtions, i have 10.

x-transition:enter="transition ease-out duration-300"
x-transition:enter-start="-translate-x-full"
x-transition:enter-end="translate-x-0"
x-transition:leave="transition ease-out duration-300"
x-transition:leave-start="translate-x-0"
x-transition:leave-end="translate-x-full"

thanks in advance

**Comment by Asim-Tahir** [comment, marked as answer]
As it says at the end of the [`x-bind`](https://alpinejs.dev/directives/bind) documentation page:

> When the directive being "bound" or "applied" is `x-for`, you should return a normal expression string from the callback. For example: `['x-for']() { return 'item in items' }
`

## Implementation
```js
Alpine.data("transitions", () => ({
  slideTransition: {
    ["x-transition:enter"]() {
      return "transition ease-out duration-300";
    },
    ["x-transition:enter-start"]() {
      return "-translate-x-full";
    },
    ["x-transition:enter-end"]() {
      return "translate-x-0";
    },
    ["x-transition:leave"]() {
      return "transition ease-out duration-300";
    },
    ["x-transition:leave-start"]() {
      return "translate-x-0";
    },
    ["x-transition:leave-end"]() {
      return "translate-x-full";
    },
  },

  ... // and other custom transitions
}));
```

## Usage
```html
<div x-data="transitions" x-bind="slideTransition">
  ...
</div>
```

**Reply by ghost** [reply]
Thanks for the response Asim.
